### PR TITLE
[docs] Add 3D reduction example to tensordot docs

### LINKED ISSUE #22748: [docs] torch.tensordot example is a little confusing
## 📚 Documentation

image

It would be good for the example to do a contraction over three dimensions instead of two. Looking at the example alone, it isn't clear if you're supposed to pass in a collection of pairs or a pair of collections. There is one line in the docs that mentions this, but making the example clearer would be nice.

### PR DESCRIPTION
**Pull request by SplitInfinity**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45697 [docs] Add 3D reduction example to tensordot docs**

**Summary**
This commit adds an example of a reduction over three dimensions with
`torch.tensordot`. It is unclear from existing docs whether `dims`
should be a list of pairs or a pair of lists.

**Test Plan**
Built the docs locally.

*Before*
Captura de Pantalla 2020-10-01 a la(s) 1 35 46 p  m

*After*
Captura de Pantalla 2020-10-05 a la(s) 12 06 28 p  m

**Fixes**
This commit closes #22748.

Differential Revision: [D24118186](https://our.internmc.facebook.com/intern/diff/D24118186)

### CODE REVIEW

**Comment by SplitInfinity** [comment]
I'm not sure why the result shows up in scientific notation but it's not consistent with the rest of the docs. I'd like to adjust that if possible.

File: torch/functional.py [hunk 1]
```diff
@@ -919,6 +919,12 @@ def tensordot(a, b, dims=2):
                 [ 3.3161,  0.0704,  5.0187, -0.4079, -4.3126,  4.8744],
                 [ 0.8223,  3.9445,  3.2168, -0.2400,  3.4117,  1.7780]])
 
+        >>> a = torch.randn(3, 5, 4, 6)
+        >>> b = torch.randn(6, 4, 5, 3)
+        >>> torch.tensordot(a, b, dims=([2, 1, 3], [1, 2, 0]))
+        tensor([[ 9.4136e+00, -1.0321e+01, -3.2591e+00],
+                [-9.8923e+00, -1.5904e+01,  7.2188e+00],
+                [ 5.1670e-03,  8.7355e+00,  1.9348e+01]])
```
Comment (zou3519) [inline torch/functional.py:927]: You can adjust https://pytorch.org/docs/stable/generated/torch.set_printoptions.html#torch.set_printoptions before printing.

File: torch/functional.py [hunk 2]
```diff
@@ -884,7 +884,7 @@ def tensordot(a, b, dims=2):
     Args:
       a (Tensor): Left tensor to contract
       b (Tensor): Right tensor to contract
-      dims (int or tuple of two lists of integers): number of dimensions to
+      dims (int or tuple of two lists of int): number of dimensions to
```
Comment (zou3519) [inline torch/functional.py:887]: nit: I think `int or Tuple[List[Int]]` makes it a clearer (all of the "of"s throw me off a little)

**Review by zou3519 — APPROVED** [review]

File: torch/functional.py (hunk 2 above)
Comment (SplitInfinity) [reply torch/functional.py]: Fixed, but I used `Tuple[List[int]] of two lists` to indicate that the tuple must be a pair.

Comment (zou3519) [reply torch/functional.py]: Sorry, I didn't read that part. `Tuple[List[int]] of two lists` sounds good to me (y)

**Pull request closed without merging**